src/nvim/os/job.c

```diff
@@ -175,8 +185,8 @@ Job *job_start(JobOptions opts, int *status)
     rstream_set_stream(job->err, job->proc_stderr);
     rstream_start(job->err);
   }
-  // Save the job to the table
-  table[i] = job;
+
+  pmap_put(int)(job_map, job->id, job);
```

Comment:
@oni-link You may use `echo 4194304 > /proc/sys/kernel/pid_max` to do this (sets max PID to 2²²). AFAIK this applies to 64-bit systems only, 32-bit have maximum limit 32768 (see `man 5 proc`, paragraph regarding pid_max file).